components/post_view/embedded_bindings/select_binding/select_binding.tsx

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {ActionResult} from 'mattermost-redux/types/actions';
+
+import {Post} from 'mattermost-redux/types/posts';
+
+import {AppBinding, AppCall} from 'mattermost-redux/types/apps';
+
+import {AppBindingLocations, AppExpandLevels} from 'mattermost-redux/constants/apps';
+
+import MenuActionProvider from 'components/suggestion/menu_action_provider';
+import AutocompleteSelector from 'components/autocomplete_selector';
+import PostContext from 'components/post_view/post_context';
+
+type Option = {
+    text: string;
+    value: string;
+};
+
+type Props = {
+    post: Post;
+    binding: AppBinding;
+    userId: string;
+    actions: {
+        doAppCall: (call: AppCall) => Promise<ActionResult>;
+    };
+};
+
+type State = {
+    selected?: Option;
+};
+
+export default class SelectBinding extends React.PureComponent<Props, State> {
+    private providers: MenuActionProvider[];
+
+    constructor(props: Props) {
+        super(props);
+
+        const binding = props.binding;
+        this.providers = [];
+        if (binding.bindings) {
+            const options = binding.bindings.map((b) => {
+                return {text: b.label, value: b.location};
+            });
+            this.providers = [new MenuActionProvider(options)];
+        }
+
+        this.state = {};
+    }
+
+    handleSelected = (selected: Option) => {
+        if (!selected) {
+            return;
+        }
+
+        this.setState({selected});
+        const binding = this.props.binding.bindings?.find((b) => b.location === selected.value);
+        if (!binding) {
+            // TODO Error?
+            return;
+        }
+
+        const {userId, post} = this.props;
+
+        const call: AppCall = {
+            url: binding.call?.url || '',
```

Comment:
I wonder if it is worth adding an AppCall constructor function somewhere to be reused.

Reply:
Added a call constructor for the calls and used here. If we move forward with this, I will create a ticket to use the constructor all around the code.

Reply:
And on mobile.